crates/sui-core/src/authority/authority_store.rs

```diff
@@ -534,6 +534,26 @@ impl AuthorityStore {
         Ok(result)
     }
 
+    pub fn check_receiving_objects(
+        &self,
+        receiving_objects: &[ObjectRef],
+        _protocol_config: &ProtocolConfig,
+    ) -> Result<(), SuiError> {
+        // TODO(tzakian): bound the number of receiving objects.
+        // Since we're at signing we check that every object reference that we are receiving is the
+        // most recent version of that object. Otherwise we return an error.
```

Comment:
can you add a TODO - when we have the marker table, we should also sign if the object has already been received, so that we can preserve liveness on signing.

Reply:
I originally had that here, but then wound up taking it out.. My thinking being that we don't need to do this at signing since IIUC the marker table is only needed for cases where there's a race around signing. 

In this case since we won't sign at all we don't need to sign if the object at that version isn't present in order to preserve liveness? Or maybe (probably) I'm missing something here...

Reply:
I realized as I was looking at this some more that you were completely right here Mark -- need to either do a range query or a point-query (like it is right now) + a lookup into the marker table. Added the TODO.